app/navigation/LocationOfGoodsNavigator.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package navigation
+
+import com.google.inject.Singleton
+import config.Constants._
+import models._
+import pages._
+import pages.locationOfGoods._
+import pages.locationOfGoods.contact.{NamePage, PhoneNumberPage}
+import play.api.mvc.Call
+
+import javax.inject.Inject
+
+@Singleton
+class LocationOfGoodsNavigator @Inject() () extends Navigator {
+
+  override def normalRoutes(departureId: String, mode: Mode): PartialFunction[Page, UserAnswers => Option[Call]] = {
+    case InferredLocationTypePage | LocationTypePage => ua => IdentificationPage.route(ua, departureId, mode)
+    case IdentificationPage                          => ua => routeIdentificationPageNavigation(ua, departureId, mode)
+    case CountryPage                                 => ua => AddressPage.route(ua, departureId, mode)
+    case MoreInformationPage                         => ua => locationOfGoodsNavigation(ua, departureId, mode)
+    case EoriPage | AuthorisationNumberPage          => ua => AddIdentifierYesNoPage.route(ua, departureId, mode)
+    case AddIdentifierYesNoPage =>
+      ua =>
+        ua.get(AddIdentifierYesNoPage) match {
+          case Some(true)  => AdditionalIdentifierPage.route(ua, departureId, mode)
+          case Some(false) => AddContactYesNoPage.route(ua, departureId, mode)
```

Comment:
This is personal preference so won't request changes but could these cases be extracted into methods, it makes it easier to read and follow the flow

Reply:
Sure thing